source/simulator/nullspace.cc

```diff
@@ -108,6 +220,56 @@ namespace aspect
   template <int dim>
   void Simulator<dim>::setup_nullspace_constraints(AffineConstraints<double> &constraints)
   {
+
+    if (parameters.nullspace_removal & Parameters<dim>::NullspaceRemoval::any_rotation)
+      {
+        const bool is_spherical = Plugins::plugin_type_matches<const GeometryModel::SphericalShell<dim>>(*geometry_model)
+                                  || Plugins::plugin_type_matches<const GeometryModel::Sphere<dim>>(*geometry_model);
+
+        Assert(is_spherical,
+               ExcNotImplemented("nullspace constraints for rotation are currently only implemented Sphere and "
+                                 "SphericalShell geometries."));
+
+        // Spherical object in 2d has 1 rotational mode and 3 in 3d. We will fix appropriate components at points on the surface.
+
+        if (dim==2)
+          {
+            // pick point p on surface (will be in positive y) and fix x direction:
+            const Point<dim> &location = geometry_model->representative_point(0.0);
+            ComponentMask mask(dof_handler.get_fe().n_components(), false);
+            mask.set(0, true);
+
+            internal::constrain_point_on_boundary(dof_handler,
+                                                  *mapping,
+                                                  location,
+                                                  mask,
+                                                  constraints);
+          }
+        else if (dim==3)
+          {
+            // point 1 is in positive z direction, constrain x and y
+            const Point<dim> &location1 = geometry_model->representative_point(0.0);
+            ComponentMask mask(dof_handler.get_fe().n_components(), false);
+            mask.set(0, true);
+            mask.set(1, true);
+            internal::constrain_point_on_boundary(dof_handler,
+                                                  *mapping,
+                                                  location1,
+                                                  mask,
+                                                  constraints);
+
+            // point 2 is in positive x direction, constrain y
+            Point<dim> location2;
+            location2[0] = location1[dim-1];
```

Comment:
This will lead to problems in deformed meshes., does it not? Maybe we need to disable this when mesh deformation is active. Or we need to rethink the approach of using known points to calculate the rotational directions.